runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillStateCache.java

```diff
@@ -79,15 +79,18 @@ public class WindmillStateCache implements StatusDataProvider {
 
   WindmillStateCache(long sizeMb, boolean supportMapViaMultimap) {
     this.workerCacheBytes = sizeMb * MEGABYTES;
+    int stateCacheConcurrencyLevel =
+        Math.max(STATE_CACHE_CONCURRENCY_LEVEL, Runtime.getRuntime().availableProcessors());
     this.stateCache =
         CacheBuilder.newBuilder()
             .maximumWeight(workerCacheBytes)
+            .weakValues()
```

Comment:
let's do the weak values separately with some more investigation.

Reply:
I didn't mean to include it. Thanks for catching.

Reply:
removed.